pkg/resources/view_grant.go

```diff
@@ -107,14 +108,24 @@ func CreateViewGrant(data *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	// ID format is <db_name>|<schema_name>|<view_name>|<privilege>
-	// view_name is empty when on_future = true
-	if futureViews {
-		data.SetId(fmt.Sprintf("%v|%v||%v", dbName, schemaName, priv))
-	} else {
-		data.SetId(fmt.Sprintf("%v|%v|%v|%v", dbName, schemaName, viewName, priv))
+	dataIdentifiers := make([][]string, 1)
+	dataIdentifiers[0] = make([]string, 4)
+	dataIdentifiers[0][0] = dbName
+	dataIdentifiers[0][1] = schemaName
+	dataIdentifiers[0][2] = viewName
+	dataIdentifiers[0][3] = priv
+
+	var buf bytes.Buffer
+	csvWriter := csv.NewWriter(&buf)
+	csvWriter.Comma = '|'
+	csvWriter.WriteAll(dataIdentifiers) // calls Flush internally
```

Comment:
```suggestion
	csvWriter.WriteAll(dataIdentifiers)
```
doesn't seem relevant/actionable to the users of this code so maybe we should not present them with this comment